keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,742 @@
-+  Existing User and existing Bucket: In this scenario, the user account has access policy defined on the `Bucket`.  The existing user privileges in the backend may conflict with the privileges that the user is requesting.
-	+ FAIL, if existing access policy is different from the requested policy.
-	+ if the existing privileges match the requested privileges, return the credentials to the workload owning the `BucketAccessRequest`.
-+ A Service Account specified and the cloud platform identity integration maps Kubernetes ServiceAccounts to the account in the backend storage system. No need to create credentials here.
-	
-Upon success, the `BucketAccess` instance is ready and the app workload can access backend storage.
+Since COSI automates the lifecycle management of credentials used for accessing buckets, and also standardizes a pattern for consuming object storage buckets in kubernetes, it is desirable to access existing buckets in a similar fashion to a bucket created by COSI. This user story explains the steps to re-use a bucket:
 
-### Static Provisioning
-"Driverless" allows the existing workloads to make use of COSI without the need for vendors to create drivers. The following steps show the details of the workflow:
-+ Admin creates `Bucket` instance that references an existing backend bucket.
-+ User creates a `BucketRequest` (BR) that references this `Bucket`.
-+ User creates `BucketAccessRequest` (BAR) that references their BR, a `BucketAccessClass`, and their secret containing access tokens.
-+ COSI detects the existence of the BAR and follows the [grant access](#grant-bucket-access) steps.
-+ COSI detects the existence of the BAR, BA and executes a validation process.
-+ As a validation step, COSI ensures that the `BucketRequest.bucketInstanceName` matches the `BucketAccessRequest.bucketAccessName.bucketInstanceName`. In other words, the two related `Bucket` references match, meaning that the BR and BAR->BA point to the same `Bucket`.
+The following stakeholders are involved in this lifecycle:
 
----
+ - Admin - creates the Bucket API object
+ - User - specifies bucket in their pod definition 
 
-# gRPC Definitions
+###### 1. Admin creates a Bucket API object
 
-There is one service defined by COSI - `Provisioner`. This will need to be satisfied by the vendor-provisioner in order to be COSI-compatible
+A Bucket with `bucketID` indicates that this Bucket has already been created outside of COSI. This bucket API object is not created as a result of a BucketRequest processed by COSI, it is rather created by the admin. 
 
 ```
-service Provisioner {
-    rpc ProvisionerGetInfo (ProvisionerGetInfoRequest) returns (ProvisionerGetInfoResponse) {}
+    Bucket - br-$uuid
+    |-------------------------------------------------|
+    | name: bucketName123                             |
+    | spec:                                           |
+    |   bucketID: bucketname123.s3.amazonaws.com      |
```

Comment:
it'd be great to keep some consistency with the other kubernetes storage concepts, from a preprovisioned volume snapshot this could be in the key `source`:

```
apiVersion: snapshot.storage.k8s.io/v1
kind: VolumeSnapshotContent
metadata:
  name: new-snapshot-content-test
spec:
  deletionPolicy: Delete
  driver: hostpath.csi.k8s.io
  source:
    snapshotHandle: 7bdd0de3-aaeb-11e8-9aae-0242ac110002
  volumeSnapshotRef:
    name: new-snapshot-test
    namespace: default
```

Reply:
I was thinking of the PV<->PVC bi-di linking